.github/workflows/maven_test.yml

```diff
@@ -18,8 +22,16 @@ jobs:
         with:
           java-version: 8
           distribution: 'temurin'
-      - name: Cache Maven packages
+      - name: Cache and restore Maven packages on master
```

Comment:
PRs might contain various dependency changes that aren't always desired in the cache. So, only write to the cache from the master branch.

PRs will still restore from the master branch, so only dependency changes within that PR need to be downloaded.